sdk/schemaregistry/azure-schemaregistry-avroencoder/azure/schemaregistry/encoder/avroencoder/_schema_registry_avro_encoder.py

```diff
@@ -205,47 +251,54 @@ def encode(
 
         if message_type:
             try:
-                return message_type.from_message_content(payload, content_type, **kwargs)
-            except AttributeError:
-                try:
-                    return message_type(payload, content_type, **kwargs)
-                except TypeError as e:
-                    SchemaEncodeError(
-                        f"""The content model {str(message_type)} is not a Callable that takes `content`
-                            and `content_type` or a subtype of the MessageType protocol.
-                            If using an Azure SDK model class, please check the README.md for the full list
-                            of supported Azure SDK models and their corresponding versions."""
-                    ).raise_with_traceback()
-
-        return {"content": payload, "content_type": content_type}
-
-    def _convert_preamble_format(
-        self, content, content_type
-    ):  # pylint: disable=no-self-use
-        record_format_identifier = b"\0\0\0\0"
-        if content[0:RECORD_FORMAT_IDENTIFIER_LENGTH] == record_format_identifier:
-            schema_id = content[
-                SCHEMA_ID_START_INDEX : (SCHEMA_ID_START_INDEX + SCHEMA_ID_LENGTH)
-            ].decode("utf-8")
-            content_type = f"{AVRO_MIME_TYPE}+{schema_id}"
-            content = content[CONTENT_START_INDEX:]
-
-        return content, content_type
+                return message_type.from_message_content(
+                    payload, content_type, **kwargs
+                )
+            except AttributeError as exc:
+                raise TypeError(
+                    f"""Cannot set content and content type on model object. The content model
+                        {str(message_type)} must be a subtype of the MessageType protocol.
+                        If using an Azure SDK model class, please check the README.md for the full list
+                        of supported Azure SDK models and their corresponding versions.""",
+                    {"content": payload, "content_type": content_type}
+                ) from exc
+
+        return MessageContent({"content": payload, "content_type": content_type})
+
+    @overload
+    def decode(
+        self,
+        message: MessageContent,
+        *,
+        readers_schema: Optional[str] = None,
+        request_options: Dict[str, Any] = None,
+        **kwargs: Any,
+    ) -> Dict[str, Any]:
+        ...
 
+    @overload
     def decode(
         self,
-        message: Union[MessageType, MessageContent],
+        message: MessageType,
         *,
         readers_schema: Optional[str] = None,
         request_options: Dict[str, Any] = None,
-        **kwargs,   # pylint: disable=unused-argument
+        **kwargs: Any,
+    ) -> Dict[str, Any]:
+        ...
```

Comment:
wondering why we have `overload` here instead of a single method signature with `message: Union[MessageType, MessageContent]`?

Reply:
ah, I think I forgot to remove this while playing around with it. Thanks for pointing it out!